neqo-transport/src/connection/tests/datagram.rs

```diff
@@ -130,24 +136,53 @@ fn limit_data_size() {
     assert!(u64::try_from(DATA_BIGGER_THAN_MTU.len()).unwrap() > DATAGRAM_LEN_MTU);
     // Datagram can be queued because they are smaller than allowed by the peer,
     // but they cannot be sent.
-    assert_eq!(client.add_datagram(DATA_BIGGER_THAN_MTU), Ok(false));
-    assert_eq!(server.add_datagram(DATA_BIGGER_THAN_MTU), Ok(false));
+    assert_eq!(client.add_datagram(DATA_BIGGER_THAN_MTU, Some(1)), Ok(()));
+    assert_eq!(server.add_datagram(DATA_BIGGER_THAN_MTU, Some(1)), Ok(()));
```

Comment:
I don't think that you need to test both client and server here (I missed this earlier).  That these use the same identifier is a bit of a warning flag.

If you do test them both like this, then moving the add_datagram() calls down to the other endpoint-specific code will make this clearer (it's easy to miss that these actions are on different endpoints).

Reply:
I will move this line and also add a comment.